### ISSUE TITLE
html2canvas not working in mobile and Ipad

### ISSUE THREAD

**Issue by anilskalyane** [niklasvh/html2canvas#1979]
This library works great in Laptop and Desktop but unable to render content on canvas in small screen devices.

Any Ideas? fix? patch?
Any help would be greatly appreciated.

--Anil

**Comment by hakimio** [comment]
By "mobile" do you mean iphone? RC1 release works well for me on Android.

**Comment by anilskalyane** [comment]
Same issues are there in both devices.

**Comment by hakimio** [comment]
You need to provide more details to get any help. Did you try RC1 release?

**Comment by anilskalyane** [comment]
It's perfectly working in windows with chrome browser but same is not working (CSS issues - attached the file) on Mac/Linux/mobile with chrome. 

[MYPdf.pdf](https://github.com/niklasvh/html2canvas/files/3546127/MYPdf.pdf)

Here is the code,

download() {
    let data = document.getElementById('contentToConvert');

    html2canvas(data, {
      allowTaint: false,
      removeContainer: true,
      backgroundColor: '#ffffff',
      scale: window.devicePixelRatio,
      useCORS: false
    }).then(canvas => {
      const contentDataURL = canvas.toDataURL('image/png')
      const imgWidth = 210;
      const pageHeight = 295;
      const imgHeight = canvas.height * imgWidth / canvas.width;
      let heightLeft = imgHeight;
      let pdf = new jspdf('p', 'mm', 'a4'); // A4 size page of PDF
      let position = 5;

      pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight)
      heightLeft -= pageHeight;

      while (heightLeft >= 0) {
        position = heightLeft - imgHeight;
        pdf.addPage();
        pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight)
        heightLeft -= pageHeight;
      }
      pdf.save('MYPdf.pdf'); // Generated PDF
    });
  }

**Comment by anilskalyane** [comment]
Finally, I got the solution. Here is my updated code,
```
download() {
    if(screen.width < 1024) {
        document.getElementById("viewport").setAttribute("content", "width=1200px");
    }
    const data = document.getElementById('contentToConvert');
    let html2canvasOptions = {
      allowTaint: true,
      removeContainer: true,
      backgroundColor: null,
      imageTimeout: 15000,
      logging: true,
      scale: 2,
      useCORS: true
    };
      // Few necessary setting options
      const contentDataURL = canvas.toDataURL('image/png')
      const imgWidth = 210;
      const pageHeight = 295;
      const imgHeight = canvas.height * imgWidth / canvas.width;
      let heightLeft = imgHeight;
      let pdf = new jspdf('p', 'mm', 'a4', true); // A4 size page of PDF
      let position = 0;

      pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST');
      heightLeft -= pageHeight;

      while (heightLeft >= 0) {
        position = heightLeft - imgHeight;
        pdf.addPage();
        pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST')
        heightLeft -= pageHeight;
      }
      pdf.save('resume.pdf'); // Generated PDF

      if(screen.width < 1024) {
        document.getElementById("viewport").setAttribute("content", "width=device-width, initial-scale=1");
      }
    });
  }
```

**Comment by estebanvr10** [comment]
Estimados les comento que yo también tuve problemas cuando hacía capture por debajo a dos div con la siguiente información:
* En el primer div tenía una gráfica de google charts.
* En el segundo div tenía un mapa que creaba con la api leaflet.
 
Los resultados que tenía eran los siguientes:
1er escenario: Cuando lo hacía desde la laptop todas las capturas se hacían sin problemas.
2do escenario: cuando lo hacía desde dispositivo android sólo podía capturar completo el segundo div, el primer div solo me guardaba el fondo. Con ios capturaba todo pero el primer div salía pixeleado y sobreescrito.

Solución: después de dos días investigando opte por cambiar los js, modifique y di formato a ambos div y cambie el javascript para que posicionara y le diera ancho y alto a la imagen antes de hacer el capture y quedó bien. 
![Screenshot_20200511_210638_com google android apps docs](https://user-images.githubusercontent.com/19638490/81642529-3a1f1500-93e9-11ea-8c0a-4a432293cb42.jpg)
![Screenshot_20200512_004200](https://user-images.githubusercontent.com/19638490/81642617-6b97e080-93e9-11ea-990b-41d16e2b1626.jpg)

Si necesitan ayuda les puedo dar la mano, eso si no tengo mucha experiencia, pero soy muy aplicado...

**Comment by RK-rocks** [comment]
> It's perfectly working in windows with chrome browser but same is not working (CSS issues - attached the file) on Mac/Linux/mobile with chrome.
> 
> [MYPdf.pdf](https://github.com/niklasvh/html2canvas/files/3546127/MYPdf.pdf)
> 
> Here is the code,
> 
> download() {
> let data = document.getElementById('contentToConvert');
> 
> ```
> html2canvas(data, {
>   allowTaint: false,
>   removeContainer: true,
>   backgroundColor: '#ffffff',
>   scale: window.devicePixelRatio,
>   useCORS: false
> }).then(canvas => {
>   const contentDataURL = canvas.toDataURL('image/png')
>   const imgWidth = 210;
>   const pageHeight = 295;
>   const imgHeight = canvas.height * imgWidth / canvas.width;
>   let heightLeft = imgHeight;
>   let pdf = new jspdf('p', 'mm', 'a4'); // A4 size page of PDF
>   let position = 5;
> 
>   pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight)
>   heightLeft -= pageHeight;
> 
>   while (heightLeft >= 0) {
>     position = heightLeft - imgHeight;
>     pdf.addPage();
>     pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight)
>     heightLeft -= pageHeight;
>   }
>   pdf.save('MYPdf.pdf'); // Generated PDF
> });
> ```
> 
> }

Hello 

your solution is working but whenever i generate pdf by this code content is cut when page divided

**Comment by RK-rocks** [comment]
> Finally, I got the solution. Here is my updated code,
> 
> ```
> download() {
>     if(screen.width < 1024) {
>         document.getElementById("viewport").setAttribute("content", "width=1200px");
>     }
>     const data = document.getElementById('contentToConvert');
>     let html2canvasOptions = {
>       allowTaint: true,
>       removeContainer: true,
>       backgroundColor: null,
>       imageTimeout: 15000,
>       logging: true,
>       scale: 2,
>       useCORS: true
>     };
>       // Few necessary setting options
>       const contentDataURL = canvas.toDataURL('image/png')
>       const imgWidth = 210;
>       const pageHeight = 295;
>       const imgHeight = canvas.height * imgWidth / canvas.width;
>       let heightLeft = imgHeight;
>       let pdf = new jspdf('p', 'mm', 'a4', true); // A4 size page of PDF
>       let position = 0;
> 
>       pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST');
>       heightLeft -= pageHeight;
> 
>       while (heightLeft >= 0) {
>         position = heightLeft - imgHeight;
>         pdf.addPage();
>         pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST')
>         heightLeft -= pageHeight;
>       }
>       pdf.save('resume.pdf'); // Generated PDF
> 
>       if(screen.width < 1024) {
>         document.getElementById("viewport").setAttribute("content", "width=device-width, initial-scale=1");
>       }
>     });
>   }
> ```

Hello your code is working but its cutting content when page is devided

**Comment by ninsau** [comment]
> Finally, I got the solution. Here is my updated code,
> 
> ```
> download() {
>     if(screen.width < 1024) {
>         document.getElementById("viewport").setAttribute("content", "width=1200px");
>     }
>     const data = document.getElementById('contentToConvert');
>     let html2canvasOptions = {
>       allowTaint: true,
>       removeContainer: true,
>       backgroundColor: null,
>       imageTimeout: 15000,
>       logging: true,
>       scale: 2,
>       useCORS: true
>     };
>       // Few necessary setting options
>       const contentDataURL = canvas.toDataURL('image/png')
>       const imgWidth = 210;
>       const pageHeight = 295;
>       const imgHeight = canvas.height * imgWidth / canvas.width;
>       let heightLeft = imgHeight;
>       let pdf = new jspdf('p', 'mm', 'a4', true); // A4 size page of PDF
>       let position = 0;
> 
>       pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST');
>       heightLeft -= pageHeight;
> 
>       while (heightLeft >= 0) {
>         position = heightLeft - imgHeight;
>         pdf.addPage();
>         pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST')
>         heightLeft -= pageHeight;
>       }
>       pdf.save('resume.pdf'); // Generated PDF
> 
>       if(screen.width < 1024) {
>         document.getElementById("viewport").setAttribute("content", "width=device-width, initial-scale=1");
>       }
>     });
>   }
> ```

You're a lifesaver!!!!! I was struggling with this until I tried your solution. Works like a charm!

**Comment by slavo3dev** [comment]
> > Finally, I got the solution. Here is my updated code,
> > ```
> > download() {
> >     if(screen.width < 1024) {
> >         document.getElementById("viewport").setAttribute("content", "width=1200px");
> >     }
> >     const data = document.getElementById('contentToConvert');
> >     let html2canvasOptions = {
> >       allowTaint: true,
> >       removeContainer: true,
> >       backgroundColor: null,
> >       imageTimeout: 15000,
> >       logging: true,
> >       scale: 2,
> >       useCORS: true
> >     };
> >       // Few necessary setting options
> >       const contentDataURL = canvas.toDataURL('image/png')
> >       const imgWidth = 210;
> >       const pageHeight = 295;
> >       const imgHeight = canvas.height * imgWidth / canvas.width;
> >       let heightLeft = imgHeight;
> >       let pdf = new jspdf('p', 'mm', 'a4', true); // A4 size page of PDF
> >       let position = 0;
> > 
> >       pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST');
> >       heightLeft -= pageHeight;
> > 
> >       while (heightLeft >= 0) {
> >         position = heightLeft - imgHeight;
> >         pdf.addPage();
> >         pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST')
> >         heightLeft -= pageHeight;
> >       }
> >       pdf.save('resume.pdf'); // Generated PDF
> > 
> >       if(screen.width < 1024) {
> >         document.getElementById("viewport").setAttribute("content", "width=device-width, initial-scale=1");
> >       }
> >     });
> >   }
> > ```
> 
> You're a lifesaver!!!!! I was struggling with this until I tried your solution. Works like a charm!

This Is really strange that this code is working.

**Comment by ninsau** [comment]
> > > Finally, I got the solution. Here is my updated code,
> > > ```
> > > download() {
> > >     if(screen.width < 1024) {
> > >         document.getElementById("viewport").setAttribute("content", "width=1200px");
> > >     }
> > >     const data = document.getElementById('contentToConvert');
> > >     let html2canvasOptions = {
> > >       allowTaint: true,
> > >       removeContainer: true,
> > >       backgroundColor: null,
> > >       imageTimeout: 15000,
> > >       logging: true,
> > >       scale: 2,
> > >       useCORS: true
> > >     };
> > >       // Few necessary setting options
> > >       const contentDataURL = canvas.toDataURL('image/png')
> > >       const imgWidth = 210;
> > >       const pageHeight = 295;
> > >       const imgHeight = canvas.height * imgWidth / canvas.width;
> > >       let heightLeft = imgHeight;
> > >       let pdf = new jspdf('p', 'mm', 'a4', true); // A4 size page of PDF
> > >       let position = 0;
> > > 
> > >       pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST');
> > >       heightLeft -= pageHeight;
> > > 
> > >       while (heightLeft >= 0) {
> > >         position = heightLeft - imgHeight;
> > >         pdf.addPage();
> > >         pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST')
> > >         heightLeft -= pageHeight;
> > >       }
> > >       pdf.save('resume.pdf'); // Generated PDF
> > > 
> > >       if(screen.width < 1024) {
> > >         document.getElementById("viewport").setAttribute("content", "width=device-width, initial-scale=1");
> > >       }
> > >     });
> > >   }
> > > ```
> > 
> > 
> > You're a lifesaver!!!!! I was struggling with this until I tried your solution. Works like a charm!
> 
> This Is really strange that this code is working.

Do you have an alternate answer you’d like to share? I’m open to new ideas to test !

**Comment by slavo3dev** [comment]
> > > > Finally, I got the solution. Here is my updated code,
> > > > ```
> > > > download() {
> > > >     if(screen.width < 1024) {
> > > >         document.getElementById("viewport").setAttribute("content", "width=1200px");
> > > >     }
> > > >     const data = document.getElementById('contentToConvert');
> > > >     let html2canvasOptions = {
> > > >       allowTaint: true,
> > > >       removeContainer: true,
> > > >       backgroundColor: null,
> > > >       imageTimeout: 15000,
> > > >       logging: true,
> > > >       scale: 2,
> > > >       useCORS: true
> > > >     };
> > > >       // Few necessary setting options
> > > >       const contentDataURL = canvas.toDataURL('image/png')
> > > >       const imgWidth = 210;
> > > >       const pageHeight = 295;
> > > >       const imgHeight = canvas.height * imgWidth / canvas.width;
> > > >       let heightLeft = imgHeight;
> > > >       let pdf = new jspdf('p', 'mm', 'a4', true); // A4 size page of PDF
> > > >       let position = 0;
> > > > 
> > > >       pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST');
> > > >       heightLeft -= pageHeight;
> > > > 
> > > >       while (heightLeft >= 0) {
> > > >         position = heightLeft - imgHeight;
> > > >         pdf.addPage();
> > > >         pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST')
> > > >         heightLeft -= pageHeight;
> > > >       }
> > > >       pdf.save('resume.pdf'); // Generated PDF
> > > > 
> > > >       if(screen.width < 1024) {
> > > >         document.getElementById("viewport").setAttribute("content", "width=device-width, initial-scale=1");
> > > >       }
> > > >     });
> > > >   }
> > > > ```
> > > 
> > > 
> > > You're a lifesaver!!!!! I was struggling with this until I tried your solution. Works like a charm!
> > 
> > 
> > This Is really strange that this code is working.
> 
> Do you have an alternate answer you’d like to share? I’m open to new ideas to test !

No i am killing my self with this shit , for over the 3 weeks, and mobile is killing me

**Comment by ninsau** [comment]
> > > > > Finally, I got the solution. Here is my updated code,
> > > > > ```
> > > > > download() {
> > > > >     if(screen.width < 1024) {
> > > > >         document.getElementById("viewport").setAttribute("content", "width=1200px");
> > > > >     }
> > > > >     const data = document.getElementById('contentToConvert');
> > > > >     let html2canvasOptions = {
> > > > >       allowTaint: true,
> > > > >       removeContainer: true,
> > > > >       backgroundColor: null,
> > > > >       imageTimeout: 15000,
> > > > >       logging: true,
> > > > >       scale: 2,
> > > > >       useCORS: true
> > > > >     };
> > > > >       // Few necessary setting options
> > > > >       const contentDataURL = canvas.toDataURL('image/png')
> > > > >       const imgWidth = 210;
> > > > >       const pageHeight = 295;
> > > > >       const imgHeight = canvas.height * imgWidth / canvas.width;
> > > > >       let heightLeft = imgHeight;
> > > > >       let pdf = new jspdf('p', 'mm', 'a4', true); // A4 size page of PDF
> > > > >       let position = 0;
> > > > > 
> > > > >       pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST');
> > > > >       heightLeft -= pageHeight;
> > > > > 
> > > > >       while (heightLeft >= 0) {
> > > > >         position = heightLeft - imgHeight;
> > > > >         pdf.addPage();
> > > > >         pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST')
> > > > >         heightLeft -= pageHeight;
> > > > >       }
> > > > >       pdf.save('resume.pdf'); // Generated PDF
> > > > > 
> > > > >       if(screen.width < 1024) {
> > > > >         document.getElementById("viewport").setAttribute("content", "width=device-width, initial-scale=1");
> > > > >       }
> > > > >     });
> > > > >   }
> > > > > ```
> > > > 
> > > > 
> > > > You're a lifesaver!!!!! I was struggling with this until I tried your solution. Works like a charm!
> > > 
> > > 
> > > This Is really strange that this code is working.
> > 
> > 
> > Do you have an alternate answer you’d like to share? I’m open to new ideas to test !
> 
> No i am killing my self with this shit , for over the 3 weeks, and mobile is killing me

Dang! And the answer above isn’t working? I sympathize. Share your code and let’s debug. It’s late where I am...so I’ll probably give you feedback in the next couple of hours. This is one reason I hate libraries

**Comment by slavo3dev** [comment]
> > > > > > Finally, I got the solution. Here is my updated code,
> > > > > > ```
> > > > > > download() {
> > > > > >     if(screen.width < 1024) {
> > > > > >         document.getElementById("viewport").setAttribute("content", "width=1200px");
> > > > > >     }
> > > > > >     const data = document.getElementById('contentToConvert');
> > > > > >     let html2canvasOptions = {
> > > > > >       allowTaint: true,
> > > > > >       removeContainer: true,
> > > > > >       backgroundColor: null,
> > > > > >       imageTimeout: 15000,
> > > > > >       logging: true,
> > > > > >       scale: 2,
> > > > > >       useCORS: true
> > > > > >     };
> > > > > >       // Few necessary setting options
> > > > > >       const contentDataURL = canvas.toDataURL('image/png')
> > > > > >       const imgWidth = 210;
> > > > > >       const pageHeight = 295;
> > > > > >       const imgHeight = canvas.height * imgWidth / canvas.width;
> > > > > >       let heightLeft = imgHeight;
> > > > > >       let pdf = new jspdf('p', 'mm', 'a4', true); // A4 size page of PDF
> > > > > >       let position = 0;
> > > > > > 
> > > > > >       pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST');
> > > > > >       heightLeft -= pageHeight;
> > > > > > 
> > > > > >       while (heightLeft >= 0) {
> > > > > >         position = heightLeft - imgHeight;
> > > > > >         pdf.addPage();
> > > > > >         pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST')
> > > > > >         heightLeft -= pageHeight;
> > > > > >       }
> > > > > >       pdf.save('resume.pdf'); // Generated PDF
> > > > > > 
> > > > > >       if(screen.width < 1024) {
> > > > > >         document.getElementById("viewport").setAttribute("content", "width=device-width, initial-scale=1");
> > > > > >       }
> > > > > >     });
> > > > > >   }
> > > > > > ```
> > > > > 
> > > > > 
> > > > > You're a lifesaver!!!!! I was struggling with this until I tried your solution. Works like a charm!
> > > > 
> > > > 
> > > > This Is really strange that this code is working.
> > > 
> > > 
> > > Do you have an alternate answer you’d like to share? I’m open to new ideas to test !
> > 
> > 
> > No i am killing my self with this shit , for over the 3 weeks, and mobile is killing me
> 
> Dang! And the answer above isn’t working? I sympathize. Share your code and let’s debug. It’s late where I am...so I’ll probably give you feedback in the next couple of hours. This is one reason I hate libraries

If you find solution let me know i am doing crazy things now  like one huge pdf 
```
export async function generate_pdf(email) {


  let data = document.getElementById("info");

  let scaleSize = 1
  let widthPDf = 600
  let heightPdf = 7000
  let pdfName = "resultsPDF"

if(screen.width < 500) {
  scaleSize = 1.5
  widthPDf = 1200
  heightPdf = 15000
  pdfName = "resultsMobile.pdf"
  }
    
await html2canvas(data,{
  scale: scaleSize,
  useCORS: false,
  imageTimeout: 20000
}).then(function(canvas) {
   // 'p', 'pt', [width,height]
   let doc = new jsPDF('p', 'pt', [widthPDf,heightPdf]);
   let imgData = canvas.toDataURL('image/jpeg');
   doc.addImage(imgData, 'JPEG',0,0,0,0,null,'FAST',0);
   doc.save(pdfName)  
})
}
```

**Comment by ninsau** [comment]
> > > > > > > Finally, I got the solution. Here is my updated code,
> > > > > > > ```
> > > > > > > download() {
> > > > > > >     if(screen.width < 1024) {
> > > > > > >         document.getElementById("viewport").setAttribute("content", "width=1200px");
> > > > > > >     }
> > > > > > >     const data = document.getElementById('contentToConvert');
> > > > > > >     let html2canvasOptions = {
> > > > > > >       allowTaint: true,
> > > > > > >       removeContainer: true,
> > > > > > >       backgroundColor: null,
> > > > > > >       imageTimeout: 15000,
> > > > > > >       logging: true,
> > > > > > >       scale: 2,
> > > > > > >       useCORS: true
> > > > > > >     };
> > > > > > >       // Few necessary setting options
> > > > > > >       const contentDataURL = canvas.toDataURL('image/png')
> > > > > > >       const imgWidth = 210;
> > > > > > >       const pageHeight = 295;
> > > > > > >       const imgHeight = canvas.height * imgWidth / canvas.width;
> > > > > > >       let heightLeft = imgHeight;
> > > > > > >       let pdf = new jspdf('p', 'mm', 'a4', true); // A4 size page of PDF
> > > > > > >       let position = 0;
> > > > > > > 
> > > > > > >       pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST');
> > > > > > >       heightLeft -= pageHeight;
> > > > > > > 
> > > > > > >       while (heightLeft >= 0) {
> > > > > > >         position = heightLeft - imgHeight;
> > > > > > >         pdf.addPage();
> > > > > > >         pdf.addImage(contentDataURL, 'PNG', 0, position, imgWidth, imgHeight, undefined,'FAST')
> > > > > > >         heightLeft -= pageHeight;
> > > > > > >       }
> > > > > > >       pdf.save('resume.pdf'); // Generated PDF
> > > > > > > 
> > > > > > >       if(screen.width < 1024) {
> > > > > > >         document.getElementById("viewport").setAttribute("content", "width=device-width, initial-scale=1");
> > > > > > >       }
> > > > > > >     });
> > > > > > >   }
> > > > > > > ```
> > > > > > 
> > > > > > 
> > > > > > You're a lifesaver!!!!! I was struggling with this until I tried your solution. Works like a charm!
> > > > > 
> > > > > 
> > > > > This Is really strange that this code is working.
> > > > 
> > > > 
> > > > Do you have an alternate answer you’d like to share? I’m open to new ideas to test !
> > > 
> > > 
> > > No i am killing my self with this shit , for over the 3 weeks, and mobile is killing me
> > 
> > 
> > Dang! And the answer above isn’t working? I sympathize. Share your code and let’s debug. It’s late where I am...so I’ll probably give you feedback in the next couple of hours. This is one reason I hate libraries
> 
> If you find solution let me know i am doing crazy things now like one huge pdf
> 
> ```
> export async function generate_pdf(email) {
> 
> 
>   let data = document.getElementById("info");
> 
>   let scaleSize = 1
>   let widthPDf = 600
>   let heightPdf = 7000
>   let pdfName = "resultsPDF"
> 
> if(screen.width < 500) {
>   scaleSize = 1.5
>   widthPDf = 1200
>   heightPdf = 15000
>   pdfName = "resultsMobile.pdf"
>   }
>     
> await html2canvas(data,{
>   scale: scaleSize,
>   useCORS: false,
>   imageTimeout: 20000
> }).then(function(canvas) {
>    // 'p', 'pt', [width,height]
>    let doc = new jsPDF('p', 'pt', [widthPDf,heightPdf]);
>    let imgData = canvas.toDataURL('image/jpeg');
>    doc.addImage(imgData, 'JPEG',0,0,0,0,null,'FAST',0);
>    doc.save(pdfName)  
> })
> }
> ```

  const downloadPdf = () => {
    if (screen.width < 1024) {
      document
        .getElementById("viewport")
        .setAttribute("content", "width=1024px");
    }
    const input = document.getElementById("info");
    html2canvas(input).then((canvas) => {
      const imgData = canvas.toDataURL("image/svg");
      const pdf = new jsPDF("landscape", "mm", "a4");
      pdf.addImage(imgData, "svg", 10, 10, 180, 0);
      pdf.save(`myfile.pdf`);
      if (screen.width < 1024) {
        document
          .getElementById("viewport")
          .setAttribute("content", "width=device-width, initial-scale=1");
      }
    });
  };



//version 2...try if first one doesnt work for you

  const downloadPDF = () => {
    if (screen.width < 1024) {
      document
        .getElementById("viewport")
        .setAttribute("content", "width=1024px");
    }
    const input = document.getElementById("info");
    html2canvas(input).then((canvas) => {
      const imgData = canvas.toDataURL("image/svg");
      const contentDataURL = canvas.toDataURL("image/png");
      const imgWidth = 210;
      const pageHeight = 295;
      const imgHeight = (canvas.height * imgWidth) / canvas.width;
      let heightLeft = imgHeight;
      let pdf = new jsPDF("p", "mm", "a4", true);
      let position = 0;
      pdf.addImage(
        imgData,
        "svg",
        0,
        position,
        imgWidth,
        imgHeight,
        undefined,
        "FAST"
      );
      heightLeft -= pageHeight;

      while (heightLeft >= 0) {
        position = heightLeft - imgHeight;
        pdf.addPage();
        pdf.addImage(
          contentDataURL,
          "PNG",
          0,
          position,
          imgWidth,
          imgHeight,
          undefined,
          "FAST"
        );
        heightLeft -= pageHeight;
      }
      pdf.save(`myfile.pdf`);
      if (screen.width < 1024) {
        document
          .getElementById("viewport")
          .setAttribute("content", "width=device-width, initial-scale=1");
      }
    });
  };



//Make sure that your viewport is set in your header woth the id required to tag it


        <meta name="viewport" content="width=device-width, initial-scale=1" id="viewport"/>

**Comment by stacklast** [comment]
If it helps anyone looking for another solution using the last version. 

- I added `windowWidth = '1280px' as fixed property. `
- Detect device `const iOS = /iPad|iPhone|iPod/.test(navigator.userAgent) && !window.MSStream;`
- Finally condition
`if (iOS) {
      const link = pdf.output("bloburl");
      setTimeout(() => {
        window.open(link, '_top');
      });
    } else {
      pdf.save(fileName +'.pdf');
    }`
I found this through testing and for some weird reason on IOS when we have more than 1 page and a half we get the pdf saved as an empty file.
